mlir/lib/Transforms/RemoveDeadValues.cpp

```diff
@@ -72,15 +72,50 @@ using namespace mlir::dataflow;
 
 namespace {
 
+// Set of structures below to be filled with operations and arguments to erase.
+// This is done to separate analysis and tree modification phases,
+// otherwise analysis is operating on half-deleted tree which is incorrect.
+
+struct CleanupFunction {
```

Comment:
To me, `CleanupFunction` is a function that _does the cleanup_. Perhaps `FunctionToCleanUp`? And then `BlockToCleanUp`, etc? Or `FunctionToUpdate` + `BlockToUpdate`?